python/dgl/dataloading/dataloader.py

```diff
@@ -351,9 +387,25 @@ def _prefetch_update_feats(
                         "Found a LazyFeature with no ID specified, "
                         "and the graph does not have dgl.NID or dgl.EID columns"
                     )
-                feats[tid, key] = get_storage_func(parent_key, type_).fetch(
-                    column.id_ or default_id, device, pin_prefetcher
-                )
+                ids = column.id_ or default_id
+                if (parent_key, type_) in gpu_caches:
+                    cache, item_shape = gpu_caches[parent_key, type_]
+                    values, missing_index, missing_keys = cache.query(ids)
+                    missing_values = get_storage_func(parent_key, type_).fetch(
+                        missing_keys, device, pin_prefetcher
+                    )
+                    cache.replace(
+                        missing_keys, F.astype(missing_values, F.float32)
+                    )
+                    values = F.astype(values, F.dtype(missing_values))
+                    F.scatter_row_inplace(values, missing_index, missing_values)
+                    F.reshape(values, (values.shape[0],) + item_shape)
```

Comment:
Why need `reshape` here?

Reply:
My understanding is it store a flatten version of the desired data, if that is the case, can we add a description about it?